src/seml/experiment/config.py

```diff
@@ -942,13 +944,38 @@ def config_get_exclude_keys(config: dict, config_unresolved: dict) -> list[str]:
     return exclude_keys
 
 
+def create_starts_with_regex(*strings: str):
```

Comment:
This whole "use a regex" to find out interpolation keys seems a bit over-complicated to me. Why not just use something like `any(x.startswith(key) for key in interpolation_keys)`? Then we wouldn't need a separate function.

Reply:
Ok, I maybe should have read the description of the PR. If we get performance, it's fine...